client/src/lib.rs

```diff
@@ -272,6 +365,28 @@ pub enum ClientInitError {
     SystemInfoCallError(tonic::Status),
 }
 
+/// Errors thrown while updating an existing client.
+#[derive(thiserror::Error, Debug)]
+pub enum ClientUpdateError {
+    /// Cannot change the channel configuration.
+    #[error("Cannot send the update: {0:?}")]
+    ControlChannelError(#[from] SendError<Change<i32, Endpoint>>),
+    /// Invalid update, cannot create the new endpoint.
+    #[error("Cannot create an endpoint: {0:?}")]
+    InvalidUpdateError(#[from] ClientInitError),
+}
+
+/// Metadata to enable the dynamic configuration of a client.
+#[derive(Clone, Debug)]
+pub struct DynamicUpdateInfo {
```

Comment:
Doesn't need to be pub

Reply:
They may need to access the changeset to get the current options when there is no mutable ref to the client.

Reply:
It's not accessible as it stands anyway. It's stored as a private field and there's no accessor.

Reply:
Good point. We need to decide whether to get rid of changeset and update ClientOptions directly (with mutex), or not, and then either add the accessor or make everything private.